source/core/Ky.ts

```diff
@@ -365,4 +383,66 @@ export class Ky {
 			},
 		);
 	}
+
+	protected _getTotalBytes(body: globalThis.BodyInit): number {
+		if (body instanceof globalThis.Blob) {
+			return body.size;
+		}
+		if (body instanceof globalThis.ArrayBuffer) {
+			return body.byteLength;
+		}
+		if (typeof body === 'string') {
+			return new globalThis.TextEncoder().encode(body).length;
+		}
+		if (body instanceof URLSearchParams) {
+			return new globalThis.TextEncoder().encode(body.toString()).length;
+		}
+		if (body instanceof globalThis.FormData) {
+			// This is an approximation, as FormData size calculation is not straightforward
+			return Array.from(body.entries()).reduce((acc, [_, value]) => {
+				if (typeof value === 'string') {
+					return acc + new globalThis.TextEncoder().encode(value).length;
+				}
+				if (value instanceof Blob) {
+					return acc + value.size;
+				}
+				return acc;
+			}, 0);
+		}
+		if ('byteLength' in body) {
+			return (body as globalThis.ArrayBufferView).byteLength;
+		}
+		return 0; // Default case, unable to determine size
+	}
+
+	protected _wrapBodyWithUploadProgress(
```

Comment:
I think it would make more sense for this function to have a similar signature to `_stream()`, i.e. take two arguments, a request and `onUploadProgress`, and return a request.

We could rename these functions `_streamRequest()` and `_streamResponse()`.

Reply:
Changed to _streamRequest and _streamResponse

Reply:
Can we merge it @sholladay?

Reply:
I haven't had sufficient time to do a proper review or to try it out. And I probably won't for about a week or so. Maybe @sindresorhus or @szmarczak can take a look, otherwise I'll get to it as soon as I can.